src/permission/config-permission-handler.js

```diff
@@ -0,0 +1,243 @@
+ * @param   {Object}   message  a parsed deepstream message
+ * @param   {Function} callback the callback to provide the result
+ * @param   {[Object]}   authData additional optional authData as passed to isValidUser
+ *
+ * @public
+ * @interface
+ * @returns {void}
+ */
+ConfigPermissionHandler.prototype.canPerformAction = function( username, message, callback, authData ) {
+	if( typeof message.data[ 0 ] !== STRING ) {
+		callback( 'invalid message', false );
+		return;
+	}
+
+	var ruleSpecification = rulesMap.getRulesForMessage( message );
+	var name = message.data[ 0 ];
+	var ruleData;
+
+	if( ruleSpecification === null ) {
+		callback( null, true );
+		return;
+	}
+
+	ruleData = this._getCompiledRulesForName( name, ruleSpecification );
+
+	if( ruleData === null ) {
+		callback( null, false );
+		return;
+	}
+
+	new RuleApplication({
+		username: username,
+		authData: authData,
+		path: ruleData,
+		ruleSpecification:ruleSpecification,
+		message: message,
+		action: ruleSpecification.action,
+		regexp: ruleData.regexp,
+		rule: ruleData.rule,
+		name:name,
+		callback: callback,
+		options: this._options
+	});
+};
+
+/**
+ * Evaluates the rules within a section and returns the matching rule for a path.
+ * Takes basic specificity (as deduced from the path length) into account and
+ * caches frequently used rules for faster access
+ *
+ * @param   {String} name              the name of the record, event or rpc the rule applies to
+ * @param   {Object} ruleSpecification a ruleSpecification as provided by the rules-map
+ *
+ * @private
+ * @returns {Object} compiled rules
+ */
+ConfigPermissionHandler.prototype._getCompiledRulesForName = function( name, ruleSpecification ) {
+
+	if( this._ruleCache.has( ruleSpecification.section, name ) ) {
+		return this._ruleCache.get( ruleSpecification.section, name );
+	}
+
+	var section = this._config[ ruleSpecification.section ];
+	var i = 0;
+	var pathLength = 0;
+	var result = null;
+
+	for( i; i < section.length; i++ ) {
+		if( section[ i ].regexp.test( name ) && section[ i ].path.length >= pathLength ) {
+			pathLength = section[ i ].path.length;
+			result = {
+				path: section[ i ].path,
+				regexp: section[ i ].regexp,
+				rule: section[ i ].rules[ ruleSpecification.type ]
+			};
+		}
+	}
+
+	if( result ) {
+		this._ruleCache.set( ruleSpecification.section, name, result );
+	}
+
+	return result;
+};
+
+/**
+ * Callback for loadConfig. Parses the incoming configuration string and forwards
+ * it to useConfig if no errors occured
+ *
+ * @param   {Error} loadError a FileSystem Error that occured during the loading of the file
+ * @param   {String} data     the content of the permission.json file as utf-8 encoded string
+ *
+ * @private
+ * @returns {void}
+ */
+ConfigPermissionHandler.prototype._onConfigLoaded = function( path, loadError, data ) {
+	if( loadError ) {
+		this.emit( 'error', 'error while loading config: ' + loadError.toString() );
```

Comment:
If a dependency errors during startup, the dependency initializer will kill the process ([code here](https://github.com/deepstreamIO/deepstream.io/blob/master/src/utils/dependency-initialiser.js#L83))

Once the plugin is ready, its errors are forwarded to the logger ([code here](https://github.com/deepstreamIO/deepstream.io/blob/master/src/deepstream.io.js#L240))